packages/ckeditor5-table-resize/src/tablecolumnresize/tablecolumnresizeediting.js

```diff
@@ -680,4 +681,24 @@ export default class TableColumnResizeEditing extends Plugin {
 			}
 		};
 	}
+
+	/**
+	 * Inserts colgroup if it is missing from table (e.g. after table insertion into table).
+	 *
+	 * @private
+	 */
+	_registerColgroupFixer() {
+		const editor = this.editor;
+
+		this.listenTo( editor.editing.view.document, 'layoutChanged', () => {
+			const table = editor.model.document.selection.getFirstPosition().findAncestor( 'table' );
+			const tableView = editor.editing.view.document.selection.getFirstPosition().getAncestors().reverse().find(
+				element => element.name === 'table'
+			);
+
+			if ( table && table.hasAttribute( 'columnWidths' ) && tableView && tableView.getChild( 0 ).name !== 'colgroup' ) {
```

Comment:
I think the hard assumption that `colgroup` element is at the index `0` at some point may prove us wrong if a client has some custom element within the `figure` - in the rest of the plugin we always operate using the `find()` method (e.g. `[ ...viewTable.getChildren() ].find(...)`) and therefore I think we should stick with it. The same applies to the unit test.